### ISSUE TITLE
Continuous contact feature

### ISSUE THREAD

**Issue by ajkluber** [markovmodel/PyEMMA#496]
Would it be possible to add a variant of the contact feature which is a continuous switching function, such as a tanh, instead of a step-function? 

For clarity I'm thinking of something like:
0.5*(tanh((r_0 - r)/w) + 1)   

Such a function would be parameterized by the the shift r_0 and the scale over which the switching occurs, w.

I am willing to attempt to implement this myself and submit a pull request if the contribution would be welcome (but would also be content if someone more familiar with the code base was interested in implementing this type of feature). I'm familiar that there is an "add_custom_feature" method to the featurizer class, would this well-suited for this particular task?

**Comment by marscher** [comment]
On 20.08.2015 02:08, Alexander Kluber wrote:

> Would it be possible to add a variant of the contact feature which is a
> continuous switching function, such as a tanh, instead of a step-function?
> Yes, I think so. You can use any mdtraj analysis function
> (compute_distances, compute_neighbours, contacts etc.) you want in your
> custom feature and transform the resulting Numpy arrays further.
> 
> For clarity I'm thinking of something like:
> 0.5*(tanh((r_0 - r)/w) + 1)
> 
> Such a function would be parameterized by the the shift r_0 and the
> scale over which the switching occurs, w.
> A continuous function has some nice properties, one could exploit during
> discretization. May I ask for which kind of system you are going to use
> this?
> 
> I am willing to attempt to implement this myself and submit a pull
> request if the contribution would be welcome (but would also be content
> if someone more familiar with the code base was interested in
> implementing this type of feature). I'm familiar that there is an
> "add_custom_feature" method to the featurizer class, would this
> well-suited for this particular task?
> Absolutely, it is exactly what you are looking for at the moment.

Unluckily we've recently found a bug in 1.2.2 in CustomFeature handling
and I'll suggest to use the devel version, since it currently only fixed
there.

If the feature works, is useful and is well documented, we are happy to
accept a PR making it available for all users.

**Comment by ajkluber** [comment]
This seemed pretty simple to implement by emulating ContactFeature, so I forked and gave it a go. It works as far as I can tell. I added the analogous test in test_featurizer as well. I'll submit a pull request shortly. 

I work with structure-based models of proteins for studying protein folding and functional conformational changes. Contact-based reaction coordinates are typically very good for these types of models by construction, but the continuous definition of a contact is interesting because it can be implemented in umbrella sampling.
